lib/enigma.rb

```diff
@@ -0,0 +1,43 @@
+class Enigma
+attr_reader :characters
+
+  def initialize(characters = ("a".."z").to_a << " ")
+    @characters = characters
+  end
+
+  def get_encoded_character(character, shift)
+    if @characters.include?(character)
+      encoded_characters = @characters.rotate(shift)
+      encoded_characters[@characters.index(character)]
+    else
+      character
+    end
+  end
+
+  def shift_message(message, shifts, encryption = [])
+    message.downcase.each_char do |character|
+      encryption << get_encoded_character(character, shifts.first)
+      shifts = shifts.rotate(1)
+    end
+    encryption.join
+  end
+
+  def encrypt(message, key, date)
+    shifts = Shift.new(key, date).total_shifts.values
+    encryption = {}
+    encryption[:encryption] = shift_message(message, shifts)
+    encryption[:key] = key
+    encryption[:date] = date
+    encryption
+  end
+
+  def decrypt(message, key, date)
+    shifts = Shift.new(key, date).total_shifts.values
+    unshifts = shifts.map{|shift_value| shift_value * -1}
+    decryption = {}
+    decryption[:decryption] = shift_message(message, unshifts)
+    decryption[:key] = key
+    decryption[:date] = date
+    decryption
+  end
```

Comment:
For the next iteration there needs to a be a way to  generate a  date and convert it to the proper format mmddyy which can then be used for the date. This should be the default value for the parameters for encrypt and decrypt.